primitives/src/v1.rs

```diff
@@ -687,7 +703,7 @@ sp_api::decl_runtime_apis! {
 			-> Option<PersistedValidationData<N>>;
 
 		/// Checks if the given validation outputs pass the acceptance criteria.
-		fn check_validation_outputs(para_id: Id, outputs: ValidationOutputs) -> bool;
+		fn check_validation_outputs(para_id: Id, outputs: ValidationOutputs) -> Result<(), CheckValidationOutputsError>;
```

Comment:
@pepyakin okay?

Reply:
The rationale behind this bool is to minimize potential effects on the runtime API: changing this struct is backward incompatible. My logic was that the detailed programmatic information about the source of the failure is not really important since it doesn't allow for recovery. It is important for diagnostics but that's better handled by logging, I guess.

Reply:
I first had logging from the runtime. However, I was not really happy with it. In the end we also have stuff like transactionvalidityerror as enum that is send over this bound. But I can also change it back.

I think it is quite important to log this stuff, otherwise it is impossible to debug all this. I just found the bug by manually adding printlns etc. And this shouldn't be the solution here ;)